substrate/client/chain-spec/src/chain_spec.rs

```diff
@@ -144,20 +161,55 @@ pub struct RawGenesis {
 	pub children_default: BTreeMap<StorageKey, GenesisStorage>,
 }
 
+impl From<sp_core::storage::Storage> for RawGenesis {
+	fn from(value: sp_core::storage::Storage) -> Self {
+		Self {
+			top: value.top.into_iter().map(|(k, v)| (StorageKey(k), StorageData(v))).collect(),
+			children_default: value
+				.children_default
+				.into_iter()
+				.map(|(sk, child)| {
+					(
+						StorageKey(sk),
+						child
+							.data
+							.into_iter()
+							.map(|(k, v)| (StorageKey(k), StorageData(v)))
+							.collect(),
+					)
+				})
+				.collect(),
+		}
+	}
+}
+
+/// Represents different formats for the GenesisConfig configuration.
 #[derive(Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 #[serde(deny_unknown_fields)]
-enum Genesis<G> {
+pub enum Genesis<G> {
+	/// (Deprecated) Contains the JSON representation of G (the native type representing the
+	/// runtime GenesisConfig struct) (will be removed with `ChainSpec::from_genesis`).
 	Runtime(G),
+	/// The genesis storage as raw data. Typically raw key-value entries in state.
 	Raw(RawGenesis),
 	/// State root hash of the genesis storage.
 	StateRootHash(StorageData),
+	/// Represents the explicit and comprehensive runtime genesis config in JSON format.
+	/// The contained object is a JSON blob that can be parsed by a compatible runtime.
+	RuntimeGenesisConfig(json::Value),
```

Comment:
```suggestion
	/// The genesis storage as raw data. Typically raw key-value entries in state.
	Raw(RawGenesis),
	/// State root hash of the genesis storage.
	StateRootHash(StorageData),
	/// Represents the explicit and comprehensive runtime genesis config in JSON format.
	/// The contained object is a JSON blob that can be parsed by a compatible runtime.
	[serde(alias = "Runtime)]
	RuntimeGenesisConfig(json::Value),
```

Shouldn't this work?

Reply:
It will. But we need to support Runtime(G) for some time, as this is public enum. And we need to support GenesisSource::Factory for a while.

Once deprecation is _enacted_, we may add this alias.